### ISSUE TITLE
Using broadcastTo

### ISSUE THREAD

**Issue by anazzani** [niivue/niivue#1052]
Hi all.

I'm working on a page containing multiple Niivue instances.

Each instance has a couple of custom sliders controlling gamma (setGamma) and crosshair position (moveCrosshairInVox).

Now I would like an option to sync the different instances and I'm exploring the broadcastTo method illustrated [here](https://niivue.github.io/niivue/features/sync.mesh.html).

It works as advertised, but not for gamma and crosshair position: I suspect this is by design but I thought I'd ask anyway in case I'm missing something obvious.

Of course I can loop through all the instances and "manually" sync myself, but I was wondering if there's a more elegant/efficient approach.

TIA.

**Comment by hanayik** [comment]
@anazzani , 

since you are moving the crosshairs programmatically, you will need to call the `nivue.sync()` function yourself in your own UI interaction handlers. 

Normally, use the mouse or touchpad to move the crosshairs will automatically call `niivue.sync()`

**Comment by anazzani** [comment]
@hanayik thanks again, but it doesn't seem to work: no errors but I see no "broadcast effect" here.

I had a quick look at your sync() function and I *think* the problem is that `this.gl.canvas.matches(":focus")` is false.

Can you confirm and, if so, is there a way to overcome the issue?

**Comment by hanayik** [comment]
@anazzani ,

Yes, good catch. Again, many of these methods are designed to work with traditional user input such as mouse events and touch events. 

In this case, you could try to programmatically set the `focus` of your desired `canvas` element in some sort of event handler when your users interact with your custom sliders that you have implemented. 

something like: 

```js
// assuming you have a niivue instances call nv1, nv2, nv3, etc.
function handleNv1SliderInput(event) {

  // set the focus state of the canvas
  nv1.gl.canvas.focus()

  // ... logic to move crosshairs ...

 // call sync()
  nv1.sync() // assuming you already registered nv2 and nv3 with nv1 so syncing works
  
  // might be good to remove the focus when you are done altering things programmatically, but could be optional
  nv1.gl.canvas.blur()
}

**Comment by anazzani** [comment]
@hanayik, works great for crosshair slider (thank you very much) but gamma slider doesn't propagate the change to other instances: is this supported by broadcastTo?

**Comment by anazzani** [comment]
Hi,

I'm still trying to figure out if there's a solution for the above mentioned issue. Does anyone have any suggestions or experiences they can share?

Thanks in advance!

**Comment by hanayik** [comment]
@anazzani , I have implemented the `gamma` syncing in the open PR #1055. You can checkout the branch referenced in that PR and run `npm run demo` then open the bidirectional sync demo on your local machine to try it. 

I'm unsure of the use case for syncing the `gamma` value across different images in niivue instances, but the option is now there to support it. This can be available in a future version of niivue once the PR is merged.
